### ISSUE TITLE
Updated support for FIL (UCL) OPM data format

### ISSUE THREAD

**Issue by neurofractal** [fieldtrip/fieldtrip#2130]
Building on #1395 I would like to contribute some code to load in data acquired from our OPM system at UCL. There have been a few cosmetic changes since 2020. Ultimately the user should be able to load in data through ```ft_preprocessing``` followed by noise reduction etc.

I am also happy to update the [tutorial script](https://www.fieldtriptoolbox.org/getting_started/opm_fil/) for the website.

**Comment by robertoostenveld** [comment]
that would be great! Could you start a pull request in draft mode, and first make only a test script that demonstrates with a small data file (not to be added to git but sent using dropbox or so) how the low-level ft_read_xxx functions are called, and that ft_preprocessing works? 

The OPM time series will be simple, have you though about how `ft_read_sens` or so would work? Usually the sensor positions are in the MEG data, but in your case they are more like how electrode positions are separate from the EEG data files.

**Comment by neurofractal** [comment]
A note to myself - need a way of switching between single and double precision. Currently hard coded.

**Comment by robertoostenveld** [comment]
> A note to myself - need a way of switching between single and double precision. Currently hard coded.

Just some tricks that I have implemented over the years in various low-level `fileio` function:

If you would store the number of channels and the number of samples (or time * fsample), you can divide the file size by the total number of channels*samples: that should be 4 (single) or 8 (double).

If you have a timestamp channel in the data (as in the CTF meg4 files), you can look at a short piece of that and see whether it increases linearly or random/jerky with either single oir double.

You could even look at a short piece of data in any channel and compare the standard deviation of the derivative over a short time segment. When you read single precision as double, or the other way around, the data will  be a mess and hence the standard deviation will be huge.

**Comment by neurofractal** [comment]
Referring to #2172 

> How do you refer to the different types of helmets or casts? I guess you have some designs that are printed in different sizes or customized to different subjects? That would be compares to an EEG cap with a 1020 electrode layout in the sizes 56cm, 58cm and 60cm.

We currently don't standardise in this way. Most sensor arrays are customised to each individual subject at the moment. I guess during import we could add grad.type = 'FIL_56_head_custom':

```FIL_56_head_custom``` where,

- FIL = institution 
- 56 = sensor number
- head = what is being scanned (we also image the spine with OPMs)
- custom = to differentiate from generic arrays when we adopt these

Importantly, this information would ideally come from the accompanying ```_meg.json``` file.  **From a BIDS perspective would this be sensible?**

**Comment by robertoostenveld** [comment]
Yes, I think that it makes sense from BIDS perspective that the meg.json would contain `HelmetManufacturer` (FIL in your case) and `HelmetManufacturersModelName` (with some string) just as for [EEG](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#example-_eegjson)
